File: frontend/src/api/payment.ts
/**
 * Payment API endpoints
 * Handles payment/topup operations for users
 */

import { apiClient } from './client'

// Types
export interface PaymentOrder {
  id: number
  trade_no: string
  user_id: number
  amount: number
  payment_amount: number
  credit_amount?: number
  status: 'pending' | 'paid' | 'expired' | 'cancelled'
  created_at: string
  paid_at?: string
  expired_at: string
  alipay_trade_no?: string
  payer_account?: string
  user?: {
    id: number
    email: string
    username?: string
  }
}

export interface PaymentConfig {
  enabled: boolean
  min_amount: number
  max_amount: number
  qr_code_url: string
  timeout_mins: number
  rate_coefficient: number  // 充值系数
}

export interface PaymentStats {
  today_order_count: number
  today_paid_count: number
  today_paid_amount: number
  total_order_count: number
  total_paid_count: number
  total_paid_amount: number
  pending_order_count: number
}

export interface MonitorStatus {
  running: boolean
  available: boolean
  last_check_time?: string
  seconds_since_check?: number
  message?: string
}

export interface PaginationResult {
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface OrderListResponse {
  items: PaymentOrder[]
  total: number
  page: number
  page_size: number
  pages: number
}

// User Payment API
export async function getConfig(): Promise<PaymentConfig> {
  const { data } = await apiClient.get<PaymentConfig>('/payment/config')
  return data
}

export async function createOrder(amount: number): Promise<PaymentOrder> {
  const { data } = await apiClient.post<PaymentOrder>('/payment/orders', { amount })
  return data
}

export async function listOrders(params?: {
  page?: number
  page_size?: number
  status?: string
}): Promise<OrderListResponse> {
  const { data } = await apiClient.get<OrderListResponse>('/payment/orders', {
    params
  })
  return data
}

export async function getOrder(tradeNo: string): Promise<PaymentOrder> {
  const { data } = await apiClient.get<PaymentOrder>(`/payment/orders/${tradeNo}`)
  return data
}

export async function cancelOrder(tradeNo: string): Promise<void> {
  await apiClient.delete(`/payment/orders/${tradeNo}`)
}

// Admin Payment API
export async function adminListOrders(params?: {
  page?: number
  page_size?: number
  status?: string
  search?: string
}): Promise<OrderListResponse> {
  const { data } = await apiClient.get<OrderListResponse>('/admin/payment/orders', {
    params
  })
  return data
}

export async function adminGetOrder(id: number): Promise<PaymentOrder> {
  const { data } = await apiClient.get<PaymentOrder>(`/admin/payment/orders/${id}`)
  return data
}

export async function adminManualConfirm(id: number): Promise<void> {
  await apiClient.post(`/admin/payment/orders/${id}/confirm`)
}

export async function adminGetStats(): Promise<PaymentStats> {
  const { data } = await apiClient.get<PaymentStats>('/admin/payment/stats')
  return data
}

export async function adminGetMonitorStatus(): Promise<MonitorStatus> {
  const { data } = await apiClient.get<MonitorStatus>('/admin/payment/monitor/status')
  return data
}

export async function adminStartMonitor(): Promise<void> {
  await apiClient.post('/admin/payment/monitor/start')
}

export async function adminStopMonitor(): Promise<void> {
  await apiClient.post('/admin/payment/monitor/stop')
}

export const paymentAPI = {
  // User APIs
  getConfig,
  createOrder,
  listOrders,
  getOrder,
  cancelOrder,

  // Admin APIs
  admin: {
    listOrders: adminListOrders,
    getOrder: adminGetOrder,
    manualConfirm: adminManualConfirm,
    getStats: adminGetStats,
    getMonitorStatus: adminGetMonitorStatus,
    startMonitor: adminStartMonitor,
    stopMonitor: adminStopMonitor
  }
}

export default paymentAPI

File: backend/internal/handler/dto/payment.go
package dto

import (
	"time"

	"github.com/Wei-Shaw/sub2api/internal/service"
)

// PaymentOrder 支付订单 DTO
type PaymentOrder struct {
	ID            int64      `json:"id"`
	TradeNo       string     `json:"trade_no"`
	UserID        int64      `json:"user_id"`
	Amount        float64    `json:"amount"`
	PaymentAmount float64    `json:"payment_amount"`
	CreditAmount  *float64   `json:"credit_amount,omitempty"`
	Status        string     `json:"status"`
	CreatedAt     time.Time  `json:"created_at"`
	PaidAt        *time.Time `json:"paid_at,omitempty"`
	ExpiredAt     time.Time  `json:"expired_at"`
	AlipayTradeNo *string    `json:"alipay_trade_no,omitempty"`
	PayerAccount  *string    `json:"payer_account,omitempty"`

	User *User `json:"user,omitempty"`
}

// PaymentOrderFromService 从 service 模型转换为 DTO
func PaymentOrderFromService(m *service.PaymentOrder) *PaymentOrder {
	if m == nil {
		return nil
	}
	out := &PaymentOrder{
		ID:            m.ID,
		TradeNo:       m.TradeNo,
		UserID:        m.UserID,
		Amount:        m.Amount,
		PaymentAmount: m.PaymentAmount,
		CreditAmount:  m.CreditAmount,
		Status:        m.Status,
		CreatedAt:     m.CreatedAt,
		PaidAt:        m.PaidAt,
		ExpiredAt:     m.ExpiredAt,
		AlipayTradeNo: m.AlipayTradeNo,
		PayerAccount:  m.PayerAccount,
	}
	if m.User != nil {
		out.User = UserFromService(m.User)
	}
	return out
}

// PaymentOrdersFromService 批量转换
func PaymentOrdersFromService(models []service.PaymentOrder) []PaymentOrder {
	out := make([]PaymentOrder, 0, len(models))
	for i := range models {
		if p := PaymentOrderFromService(&models[i]); p != nil {
			out = append(out, *p)
		}
	}
	return out
}

// PaymentConfig 支付配置 DTO
type PaymentConfig struct {
	Enabled         bool    `json:"enabled"`
	MinAmount       float64 `json:"min_amount"`
	MaxAmount       float64 `json:"max_amount"`
	QRCodeURL       string  `json:"qr_code_url"`
	TimeoutMins     int     `json:"timeout_mins"`
	RateCoefficient float64 `json:"rate_coefficient"` // 充值系数
}

// PaymentConfigFromService 从 service 响应转换为 DTO
func PaymentConfigFromService(m *service.PaymentConfigResponse) *PaymentConfig {
	if m == nil {
		return nil
	}
	return &PaymentConfig{
		Enabled:         m.Enabled,
		MinAmount:       m.MinAmount,
		MaxAmount:       m.MaxAmount,
		QRCodeURL:       m.QRCodeURL,
		TimeoutMins:     m.TimeoutMins,
		RateCoefficient: m.RateCoefficient,
	}
}

// PaymentStats 支付统计 DTO
type PaymentStats struct {
	TodayOrderCount   int64   `json:"today_order_count"`
	TodayPaidCount    int64   `json:"today_paid_count"`
	TodayPaidAmount   float64 `json:"today_paid_amount"`
	TotalOrderCount   int64   `json:"total_order_count"`
	TotalPaidCount    int64   `json:"total_paid_count"`
	TotalPaidAmount   float64 `json:"total_paid_amount"`
	PendingOrderCount int64   `json:"pending_order_count"`
}

// PaymentStatsFromService 从 service 模型转换为 DTO
func PaymentStatsFromService(m *service.PaymentStats) *PaymentStats {
	if m == nil {
		return nil
	}
	return &PaymentStats{
		TodayOrderCount:   m.TodayOrderCount,
		TodayPaidCount:    m.TodayPaidCount,
		TodayPaidAmount:   m.TodayPaidAmount,
		TotalOrderCount:   m.TotalOrderCount,
		TotalPaidCount:    m.TotalPaidCount,
		TotalPaidAmount:   m.TotalPaidAmount,
		PendingOrderCount: m.PendingOrderCount,
	}
}
